### PR TITLE
Add some extra track-only related process types

### PR DESCRIPTION
**Pull request by tdealtry**
Some track processes were set to `kUnknownProcess` in the output file. Explicitly add them in

Also made having `kUnknownProcess` not allowed via the string->enum function, and info about how to contribute when people find new processes that aren't defined (I'm sure there'll be some!)

### CODE REVIEW

**Comment by nickwp** [comment]
In case it's useful... here's a list of process names I found by grepping G4 source code. It's probably incomplete, has some duplicates, and obviously has some that are irrelevant, but maybe useful beyond just waiting for a process to occur and give an error before adding them?

G4MuonMinusAtomicCapture: `muMinusAtomicCaptureAtRest`
G4HadronStoppingProcess: `hadronCaptureAtRest`
G4eplusPolarizedAnnihilation: `pol-annihil`
G4ePolarizedIonisation: `pol-eIoni`
G4ePolarizedBremsstrahlung: `pol-eBrem`
G4eeToHadrons: `ee2hadr`
G4hhIonisation: `hhIoni`
G4eBremsstrahlung: `eBrem`
G4ionIonisation: `ionIoni`
G4alphaIonisation: `alphaIoni`
G4eIonisation: `eIoni`
G4hIonisation: `hIoni`
G4CoulombScattering: `CoulombScat`
G4eplusAnnihilation: `annihil`
G4MuIonisation: `muIoni`
G4HadronicProcess: `Hadronic`
G4RadioactiveDecay: `RadioactiveDecay`
G4hMultipleScattering: `msc`
G4NuclearStopping: `nuclearStopping`
G4AdjointhMultipleScattering: `msc`
G4ForwardXrayTR: `XrayTR`
G4MuMultipleScattering: `muMsc`
G4DecayWithSpin: `DecayWithSpin`
G4UnknownDecay: `UnknownDecay`
G4PionDecayMakeSpin: `Decay`
G4MuonicAtomDecay: `MuonicAtomDecay`
G4Decay: `Decay`
G4StepLimiter: `StepLimiter`
G4NeutronKiller: `nKiller`
G4UserSpecialCuts: `UserSpecialCut`
G4HadronicProcess: `Hadronic`
G4RadioactiveDecay: `RadioactiveDecay`
G4ChargeExchangeProcess: `chargeExchange`
G4AntiProtonInelasticProcess: `anti_protonInelastic`
G4He3InelasticProcess: `he3Inelastic`
G4IonInelasticProcess: `ionInelastic`
G4SigmaMinusInelasticProcess: `sigma-Inelastic`
G4LambdaInelasticProcess: `lambdaInelastic`
G4AntiAlphaInelasticProcess: `anti_alphaInelastic`
G4SigmaPlusInelasticProcess: `sigma+Inelastic`
G4UCNLoss: `UCNLoss`
G4HadronElasticProcess: `hadElastic`
G4HadronFissionProcess: `nFission`
G4UCNBoundaryProcess: `UCNBoundaryProcess`
G4AntiOmegaMinusInelasticProcess: `anti_omega-Inelastic`
G4KaonZeroSInelasticProcess: `kaon0SInelastic`
G4TritonInelasticProcess: `tInelastic`
G4KaonMinusInelasticProcess: `kaon-Inelastic`
G4MuonNuclearProcess: `muonNuclear`
G4ProtonInelasticProcess: `protonInelastic`
G4AntiLambdaInelasticProcess: `anti-lambdaInelastic`
G4PhotoCaptureProcess: `photonCapture`
G4HadronCaptureProcess: `nCapture`
G4UCNMultiScattering: `UCNMultiScattering`
G4XiZeroInelasticProcess: `xi0Inelastic`
G4PionMinusInelasticProcess: `pi-Inelastic`
G4KaonPlusInelasticProcess: `kaon+Inelastic`
G4PionPlusInelasticProcess: `pi+Inelastic`
G4PhotoFissionProcess: `photonFission`
G4UCNAbsorption: `UCNAbsorption`
G4ElectronNuclearProcess: `electronNuclear`
G4PositronNuclearProcess: `positronNuclear`
G4AntiSigmaMinusInelasticProcess: `anti_sigma-Inelastic`
G4AntiTritonInelasticProcess: `anti_tritonInelastic`
G4AntiHe3InelasticProcess: `anti_He3Inelastic`
G4AntiXiZeroInelasticProcess: `anti_xi0Inelastic`
G4AntiNeutronInelasticProcess: `anti_neutronInelastic`
G4NeutronInelasticProcess: `neutronInelastic`
G4AntiDeuteronInelasticProcess: `anti_deuteronInelastic`
G4XiMinusInelasticProcess: `xi-Inelastic`
G4KaonZeroLInelasticProcess: `kaon0LInelastic`
G4DeuteronInelasticProcess: `dInelastic`
G4AntiXiMinusInelasticProcess: `anti_xi-Inelastic`
G4AntiSigmaPlusInelasticProcess: `anti_sigma+Inelastic`
G4PhotoNuclearProcess: `photonNuclear`
G4OmegaMinusInelasticProcess: `omega-Inelastic`
G4AlphaInelasticProcess: `alphaInelastic`
G4AntiNeutronAnnihilationAtRest: `AntiNeutronAnnihilationAtRest`
G4OpMieHG: `OpMieHG`
G4OpWLS: `OpWLS`
G4OpAbsorption: `OpAbsorption`
G4OpRayleigh: `OpRayleigh`
G4OpBoundaryProcess: `OpBoundary`
G4hImpactIonisation: `hImpactIoni`
G4PolarizedGammaConversion: `pol-conv`
G4PolarizedPhotoElectricEffect: `pol-phot`
G4PolarizedCompton: `pol-compt`
G4MicroElecInelastic: `MicroElecIonisation`
G4MuElecInelastic: `MuElecIonisation`
G4MuElecElastic: `MuElecElastic`
G4RayleighScattering: `Rayl`
G4MicroElecElastic: `MicroElecElastic`
G4AnnihiToMuPair: `AnnihiToMuPair`
G4hPairProduction: `hPairProd`
G4hBremsstrahlung: `hBrems`
G4eMultipleScattering: `msc`
G4ComptonScattering: `compt`
G4hMultipleScattering: `msc`
G4PhotoElectricEffect: `phot`
G4NuclearStopping: `nuclearStopping`
G4GammaConversion: `conv`
G4eAdjointMultipleScattering: `msc`
G4AdjointhMultipleScattering: `msc`
G4VTransitionRadiation: `TR`
G4ForwardXrayTR: `XrayTR`
G4TransitionRadiation: `TR`
G4Scintillation: `Scintillation`
G4Cerenkov: `Cerenkov`
G4LEPTSAttachmentModel: `G4LEPTSAttachmentModel`
G4DNAChargeIncrease: `DNAChargeIncrease`
G4DNAIonisation: `DNAIonisation`
G4DNAAttachment: `DNAAttachment`
G4DNAMolecularDissociation: `DNAMolecularDecay`
G4DNARotExcitation: `G4DNARotExcitation`
G4DNADissociation: `G4DNADissociation`
G4DNAPositronium: `G4DNAPositronium`
G4DNAElastic: `DNAElastic`
G4DNAExcitation: `DNAExcitation`
G4DNAChargeDecrease: `DNAChargeDecrease`
G4DNAVibExcitation: `DNAVibrationalExcitation`
G4ErrorEnergyLoss: `G4ErrorEnergyLoss`
G4MuMultipleScattering: `muMsc`
G4MuBremsstrahlung: `muBrems`
G4MuPairProduction: `muPairProd`
G4ePairProduction: `ePairProd`
G4PhononDownconversion: `phononDownconversion`
G4PhononScattering: `phononScattering`
G4PhononReflection: `phononReflection`
G4FastSimulationManagerProcess: `G4FastSimulationManagerProcess`
G4MuonMinusCapture: `muMinusCaptureAtRest`
G4HadronicAbsorptionFritiof: `hFritiofCaptureAtRest`
G4HadronicAbsorptionBertini: `hBertiniCaptureAtRest`

File: src/WCSimEnumerations.cc [hunk 1]
```diff
@@ -143,6 +143,39 @@ std::string WCSimEnumerations::EnumAsString(ProcessType_t p)
   case (kNeutronInelastic) :
     return "neutronInelastic";
     break;
+  case (kMuMinusCaptureAtRest) :
+    return "kMuMinusCaptureAtRest";
```
Comment (nickwp) [inline src/WCSimEnumerations.cc]: Should this return "muMinusCaptureAtRest" not "kMuMinusCaptureAtRest"?

Comment (tdealtry) [reply src/WCSimEnumerations.cc]: Good eyes. Thanks @nickwp

**Comment by tdealtry** [comment]
Thanks for the grepping magic @nickwp I've implemented them all (except the DNA stuff - hopefully there's not so much life in our water that we need to simulate the microbes...)

Hopefully this'll pass the CI now. If it does, I'll merge

**Pull request merged**